android/gradle/version.gradle

```diff
@@ -0,0 +1,39 @@
+// This script will set the app version by either appending the git short hash (if build locally)
+// or the jenkins build number (if build by Jenkins CI) to the original version name.
+// To use this script, define `projectVersion` and then apply this file.
+// example:
+// ext.projectVersion = "1.12"
+// apply from: 'version.gradle'
+//
+// the main `build.gradle` file should delete the `versionName` line under the android
+// defaultConfig block
+
+android {
+    defaultConfig {
+        versionName = projectVersion + versionSuffix()
```

Comment:
Maybe a better way to do this is just to define the methods and give an example rather than actually modifying versionName?  This would remove the need for the extra "projectVersion" variable and allow the user to integrate the information in whatever way is most natural for them.